docs/src/pages/components/view/ViewPropControls.tsx

```diff
@@ -0,0 +1,110 @@
+import * as React from 'react';
+import { Flex, ViewProps, SelectField, TextField } from '@aws-amplify/ui-react';
+import _ from 'lodash';
+
+export interface ViewPropControlsProps extends ViewProps {
+  setAriaLabel: (value: React.SetStateAction<ViewProps['ariaLabel']>) => void;
+  setWidth: (value: React.SetStateAction<ViewProps['width']>) => void;
+  setHeight: (value: React.SetStateAction<ViewProps['height']>) => void;
+  setColor: (value: React.SetStateAction<ViewProps['color']>) => void;
+  setBackgroundColor: (
+    value: React.SetStateAction<ViewProps['backgroundColor']>
+  ) => void;
+  setBoxShadow: (value: React.SetStateAction<ViewProps['boxShadow']>) => void;
+  setPadding: (value: React.SetStateAction<ViewProps['padding']>) => void;
+  setBorder: (value: React.SetStateAction<ViewProps['border']>) => void;
+  setBorderRadius: (
+    value: React.SetStateAction<ViewProps['borderRadius']>
+  ) => void;
+  setAsElementType: (value: React.SetStateAction<ViewProps['as']>) => void;
+}
+
+export interface ViewPropControlsInterface {
+  (props: ViewPropControlsProps): JSX.Element;
+}
+
+export const ViewPropControls: ViewPropControlsInterface = ({
+  ariaLabel,
+  as: asElementType,
+  backgroundColor,
+  border,
+  borderRadius,
+  boxShadow,
+  color,
+  height,
+  padding,
+  setAriaLabel,
+  setAsElementType,
+  setBackgroundColor,
+  setBorder,
+  setBorderRadius,
+  setBoxShadow,
+  setColor,
+  setHeight,
+  setPadding,
+  setWidth,
+  width,
+}) => {
+  return (
+    <Flex direction="column">
+      <SelectField
+        value={asElementType as string}
+        label="As element type"
+        onChange={(event) =>
+          setAsElementType(event.target.value as ViewProps['as'])
+        }
+      >
+        <option value="div">div</option>
+        <option value="button">button</option>
+        <option value="p">p</option>
+        <option value="span">span</option>
+      </SelectField>
+      <TextField
+        label="Aria Label"
+        value={ariaLabel}
+        onChange={(event) => setAriaLabel(event.target.value)}
+      />
+      <TextField
+        value={backgroundColor as string}
+        onChange={(event) => setBackgroundColor(event.target.value)}
+        label="Background Color"
+      />
+      <TextField
+        value={border as string}
+        onChange={(event) => setBorder(event.target.value)}
+        label="Border"
+      />
+      <TextField
+        value={borderRadius as string}
+        onChange={(event) => setBorderRadius(event.target.value)}
+        label="Border Radius"
+      />
+      <TextField
+        value={boxShadow as string}
+        onChange={(event) => setBoxShadow(event.target.value)}
+        label="Box Shadow"
+      />
+      <TextField
+        value={color as string}
+        onChange={(event) => setColor(event.target.value)}
+        label="Color"
+      />
+      <TextField
```

Comment:
I saw that @0618 was using a `SliderField` for numeric props in her TextField PR (see [FontWeight](https://github.com/aws-amplify/amplify-ui/pull/1711) in demo). Is this something we'd like to explore for props like height, borderRadius, padding, width, etc?

Reply:
That makes sense for `FontWeight` because it doesn't have a unit. I think we would just want a TextField here because a user may want to experiment with `px`, `em`, `rem`, etc. We also probably don't want to limit the range of numbers customers can put in, which would have to be decided when using a `SliderField`.